Enterprise-CMCS/managed-care-review
# [MCR-3547] Add view for adding a mccrs record number

## Summary
Adds a form for CMS users to fill out to update the MC-CRS record number which is a field on the contract table

[figma designs](https://www.figma.com/file/frKNnm6lkcpdjJ2eVJE8Bj/Managed-Care-Review?node-id=5626%3A123820&mode=dev)

#### Related issues
https://qmacbis.atlassian.net/browse/MCR-3547

#### To Do items

- [x] Update the header banner on the Add MC-CRS ID page to display the submission name. Currently it just says CMS

#### Screenshots
<img width="685" alt="Screenshot 2023-10-20 at 10 51 53 AM" src="https://github.com/Enterprise-CMCS/managed-care-review/assets/67110378/fb7ff2c2-f543-42e9-bfe7-98c29e199983">
<img width="573" alt="Screenshot 2023-10-20 at 10 52 59 AM" src="https://github.com/Enterprise-CMCS/managed-care-review/assets/67110378/0051e996-9227-48c6-9877-ab59f41fbed5">
<img width="506" alt="Screenshot 2023-10-20 at 10 53 27 AM" src="https://github.com/Enterprise-CMCS/managed-care-review/assets/67110378/cc7155ed-5d1c-40c0-9c4c-25626d70f1b6">


#### Test cases covered

- Validations for the MC-CRS record number (must be 4 digits exactly, a number, not blank)
- The add MC-CRS record number link is only displayed to CMS users

## QA guidance

- Create a submission as a state user
- View the submission summary as a CMS user
- Click "Add MC-CRS record number" link
- You should be able to enter a 4 digit number and click save to be taken back to the submission summary page
- Once back on the submission summary page you should see: MC-CRS record number:[the number you entered]
